src/v/kafka/server/request_context.h

```diff
@@ -248,12 +254,71 @@ class request_context {
         return _conn->server().credentials();
     }
 
+    // Set force to true to always enqueue the audit event regardless of event
+    // type
+    template<typename T = model::topic>
+    bool audit(
+      security::audit::event_type event_type,
+      std::vector<T> additional_resources = {}) {
+        auto operation_name = handler_for_key(_header.key).value()->name();
+        auto& audit_mgr = _conn->server().audit_mgr();
+        auto additional_resource_details
+          = security::audit::create_resource_details(additional_resources);
+        _authz_results.erase(
+          std::remove_if(
+            _authz_results.begin(),
+            _authz_results.end(),
+            [this,
+             operation_name,
+             &audit_mgr,
+             event_type,
+             &additional_resource_details](const auto& val) {
+                auto api_act = security::audit::make_api_activity_event(
+                  operation_name,
+                  val,
+                  _conn->local_address(),
+                  _conn->server().name(),
+                  _conn->client_host(),
+                  _conn->client_port(),
+                  _header.client_id,
+                  additional_resource_details);
+                if (request_contains_audit_topic()) {
+                    return audit_mgr.enqueue_mandatory_audit_event(
+                      std::move(api_act));
+                } else {
+                    return audit_mgr.enqueue_audit_event(
+                      event_type, std::move(api_act));
+                }
+            }),
+          _authz_results.end());
+
+        if (!_authz_results.empty()) {
```

Comment:
Could it ever be the case that `_authz_results` is empty to begin with? Maybe theres a handler that doesn't call `authorized()` not sure

Reply:
Are you asking to short circuit the audit call at the beginning, like

```c++
if (_authz_results.empty()) { return true; }
```
?

Reply:
Mistake, misread the condition, resolving